esmvalcore/_recipe.py

```diff
@@ -352,82 +352,85 @@ def _get_default_settings(variable, config_user, derive=False):
 def _add_fxvar_keys(fx_var_dict, variable):
     """Add keys specific to fx variable to use get_input_filelist."""
     fx_variable = dict(variable)
+    fx_variable.update(fx_var_dict)
 
     # set variable names
     fx_variable['variable_group'] = fx_var_dict['short_name']
-    fx_variable['short_name'] = fx_var_dict['short_name']
 
-    # specificities of project
+    # add special ensemble for CMIP5 only
     if fx_variable['project'] == 'CMIP5':
-        fx_variable['mip'] = 'fx'
         fx_variable['ensemble'] = 'r0i0p0'
-    elif fx_variable['project'] == 'CMIP6':
-        fx_variable['grid'] = variable['grid']
-        if 'mip' in fx_var_dict:
-            fx_variable['mip'] = fx_var_dict['mip']
-    elif fx_variable['project'] in ['OBS', 'OBS6', 'obs4mips']:
-        fx_variable['mip'] = 'fx'
+
     # add missing cmor info
     _add_cmor_info(fx_variable, override=True)
+
     return fx_variable
 
 
-def _get_correct_fx_file(variable, fx_varname, config_user):
+def _get_correct_fx_file(variable, fx_variable, config_user):
```

Comment:
My apologies for asking this late, but this is something that has been bothering me for some time. Why is this function called `_get_correct_fx_file`? Is there also a function `_get_wrong_fx_file`? Maybe `_get_fx_file` would be sufficient?

Reply:
I was planning on plugging in the suggestions from you and @JaroCamphuijsen tomorrow but this did crack me up, no, it's just simply coz I like orde, there is no wrong file getter :rofl:

Reply:
it is now, the incorrect, `_get_fx_file` :rofl: